src/lighthouse.browserspec.ts

```diff
@@ -0,0 +1,50 @@
+/** @jest-environment puppeteer */
+import * as chromeLauncher from 'chrome-launcher';
+import lighthouse from 'lighthouse';
+import { url } from './test/browserutils';
+
+const TEST_PAGE = '/books/testbook1-shortid/pages/testpage1-shortid';
+
+describe('Lighthouse audits', () => {
+  let chrome: chromeLauncher.LaunchedChrome | undefined;
+
+  const checkLighthouse = async(urlPath: string, accessibiiltyThreshold: number = 1.0) => {
+    if (!chrome) {
+      return expect(chrome).toBeTruthy();
+    }
+
+    const { lhr } = await lighthouse(url(urlPath), {port: chrome.port}, null);
+
+    expect(lhr.categories.accessibility.score).toBeGreaterThanOrEqual(accessibiiltyThreshold);
+    expect(lhr.categories.seo.score).toBeGreaterThanOrEqual(0.8);
+    expect(lhr.categories.pwa.score).toBeGreaterThanOrEqual(0.5);
+    expect(lhr.categories['best-practices'].score).toBeGreaterThanOrEqual(0.93);
+    // This one depends on how fast chrome executes so maybe we should drop it
+    expect(lhr.categories.performance.score).toBeGreaterThanOrEqual(0.4);
+  };
+
+  beforeAll(async() => {
+
+    chrome = await chromeLauncher.launch({
+      chromeFlags: ['--headless'],
+    });
+
+    // Connect to puppeteer
+    // const resp = await fetch(`http://localhost:${chrome.port}/json/version`);
+    // const { webSocketDebuggerUrl } = await resp.json();
+    // browser = await puppeteer.connect({browserWSEndpoint: webSocketDebuggerUrl});
+  });
+
+  afterAll(async() => {
+    return chrome && await chrome.kill();
+  });
+
+  it('reports about the root page', async() => {
+    await checkLighthouse('/');
+  });
+
+  it('reports about the test content page', async() => {
+    await checkLighthouse(TEST_PAGE, 0.95);
```

Comment:
I would like to see what the report generated & would like to try to get an HTML report out (similar to [tutor/scripts/accessibility-audit/index.js](https://github.com/openstax/tutor-js/blob/b7b9cc4a69d547ce7773906dde8279ad82743466/tutor/scripts/accessibility-audit/index.js) ) but maybe running the audit in the browser is sufficient for now. Thoughts?

Reply:
yea i don't think we have to worry about the html reports for now, we might generate those separately later but it wouldn't be part of the unit test suite